src/OrchardCore.Modules/OrchardCore.Media/Services/RemoteMediaCacheBackgroundTask.cs

```diff
@@ -50,12 +49,6 @@ public async Task DoWorkAsync(IServiceProvider serviceProvider, CancellationToke
             return;
         }
 
-        // Ensure that a remote media cache has been registered.
-        if (serviceProvider.GetService<IMediaFileStoreCache>() is null)
-        {
-            return;
-        }
-
```

Comment:
Such checks, also in src/OrchardCore.Modules/OrchardCore.Media/Startup.cs, need to remain, because we don't want to spin wheels needlessly to interact with a service that does nothing.

Reply:
It will be there all the time

Reply:
The code after this still shouldn't run if the null service is registered.

Also, please test if all of these still work properly with the Azure and AWS media.

Reply:
> The code after this still shouldn't run if the null service is registered.

Why

> Also, please test if all of these still work properly with the Azure and AWS media.

Unfortunately, I mightn't be able to test this against Azure or AWS but I will see if I can use sort of `FileStore`

Reply:
The code after `return` does something that's non-sense if there's no actual provider registered. Because of this, there was this `if`. You mustn't remove the `if`.